Mathlib/LinearAlgebra/Matrix/Module.lean

```diff
@@ -0,0 +1,75 @@
+/-
+Copyright (c) 2025 Yunzhou Xie. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yunzhou Xie
+-/
+import Mathlib.Algebra.Module.BigOperators
+import Mathlib.Data.Matrix.Mul
+import Mathlib.LinearAlgebra.Pi
+
+/-!
+# Main Results
+
+- `Matrix.matrixModule`: This instance shows `ι → M` is a module over `Matrix ι ι R`, and the action
+  of it is a generalization of `Matrix.mulVec`.
+- `LinearMap.mapModule`: This defines a linear map from `ι → M` to `ι → N` over
+  `Matrix ι ι R` induced by a linear map from `M` to `N` and together with `Matrix.matrixModule`
+  it gives a functor from the category of `R`-modules to the category of `Matrix ι ι R`-modules.
+
+## Tags
+matrix, module
+-/
+
+variable {ι R M N P : Type*} [Ring R] [Fintype ι] [DecidableEq ι] [AddCommGroup M] [Module R M]
+  [AddCommGroup N] [Module R N] [AddCommGroup P] [Module R P]
+
+namespace Matrix
+
+/-- `Mⁿ` is a `Mₙ(R)` module, note that this creates a diamond when `M` is `Matrix ι ι R` or when
+  `M` is `R`. -/
+scoped instance matrixModule : Module (Matrix ι ι R) (ι → M) where
+  smul N v i := ∑ j : ι, N i j • v j
+  one_smul v := funext fun i ↦ show ∑ _, _ = _ by simp [one_apply]
+  mul_smul N₁ N₂ v := funext fun i ↦ show ∑ _, _ = ∑ _, _ • (∑ _, _) by
+    simp_rw [mul_apply, Finset.smul_sum, Finset.sum_smul, MulAction.mul_smul]
+    rw [Finset.sum_comm]
+  smul_zero v := funext fun i ↦ show ∑ _, _ = _ by simp
+  smul_add N v₁ v₂ := funext fun i ↦ show ∑ j : ι, N i j • (v₁ + v₂) j = (∑ _, _) + (∑ _, _) by
+    simp [smul_add, Finset.sum_add_distrib]
+  add_smul N₁ N₂ v := funext fun i ↦ show ∑ j : ι, (N₁ + N₂) i j • v j = (∑ _, _) + (∑ _, _) by
+    simp [add_smul, Finset.sum_add_distrib]
+  zero_smul v := funext fun i ↦ show ∑ _, _ = _ by simp
+
+lemma smul_vec_def (N : Matrix ι ι R) (v : ι → M) :
```

Comment:
```suggestion
lemma smulVec_def (N : Matrix ι ι R) (v : ι → M) :
```
And can you add in the docstring of `Matrix.matrixModule` (which should probably be called `Matrix.vectorModule` or `Matrix.piModule` etc) that the intended name for the action is `smulVec`?